Source/bmalloc/libpas/src/libpas/pas_page_config_size.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef PAS_PAGE_CONFIG_SIZE_H
+#define PAS_PAGE_CONFIG_SIZE_H
+
+#include "pas_utils.h"
+
+PAS_BEGIN_EXTERN_C;
+
+enum pas_page_config_size {
```

Comment:
nit: since this isn't really the size of the page-config per se, but the objects allocated underneath it, could it have a different name? Perhaps even 'pas_object_size_class' since it's supposed to be 1-to-1 with that concept (currently just macros)

Reply:
I think maybe `pas_page_config_size_category` is good? I like keeping it associated with the `page_config` because it's not really any individual object's size either - small/medium/marge/large is decided at the page level. But you're right that it's not really the size per se.

Reply:
Resolved in latest push.